rllib/core/rl_module/rl_module.py

```diff
@@ -378,33 +401,126 @@ def get_state(self) -> Mapping[str, Any]:
     def set_state(self, state_dict: Mapping[str, Any]) -> None:
         """Sets the state dict of the module."""
 
-    def serialize(self) -> Mapping[str, Any]:
-        """Return the serialized state of the module."""
-        return {
-            "class": self.__class__,
-            "config": self.config.to_dict(),
-            "state": self.get_state(),
-        }
+    def _save_module_metadata(
+        self,
+        checkpoint_dir: Union[str, pathlib.Path],
+        module_state_path: Union[str, pathlib.Path],
+    ):
+        """Saves the metadata of the module to checkpoint_dir.
+
+        Includes:
+            - module class path
+            - module state path
+            - the module config
+            - the ray version used
+            - the ray commit hash used
+            - the date and time of the checkpoint was created
+
+        """
+        if isinstance(checkpoint_dir, str):
+            checkpoint_dir = pathlib.Path(checkpoint_dir)
+        if isinstance(module_state_path, str):
+            module_state_path = pathlib.Path(module_state_path)
+        gmt_time = datetime.datetime.utcnow().strftime("%Y-%m-%d %H:%M:%S GMT")
+        metadata = {}
+        # TODO (Avnishn): Find a way to incorporate the tune registry here.
+        metadata["module_class"] = serialize_type(self.__class__)
+        metadata["module_config"] = self.config.to_dict()
```

Comment:
how do you know self.config is json serializable?
we have been using pickle for this right?

Reply:
Honestly this is where things get confusing for me as well. I wrote in the docstring of the config object that the dict has to be json serializable and users can check this by calling json.dumps on their (config.to_dict)